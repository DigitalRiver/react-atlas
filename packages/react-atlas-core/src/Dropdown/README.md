Update Dropdown props. just for example, need to remove afterwards:

    initialState = {value: 'bird', optionsValue: "bird"}
    handleChangeProps = () => {
      setState({ value: 'cow', optionsValue: "raymans" });
    };

    <div>
        <Button primary onClick={handleChangeProps}>Change Props</Button>
          <p>state's value:{state.value}, optionsValue: {state.optionsValue}</p>
      <Dropdown value={state.value}>
        <span value={state.optionsValue}>{state.optionsValue}</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
      </Dropdown>
    </div>

Basic Dropdown:

    <div>
      <Dropdown>
            <span value="bird">Bird</span>
            <span value="cow">Cow</span>
            <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
            <span value="monkey">Monkey</span>
            <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
      </Dropdown>
    </div>

Dropdown with Label

    <Dropdown
      customLabel="Dropdown Example">
        <span value="bird">Bird</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with Label on the Left:

      <Dropdown customLabel="Dropdown Example:" leftLabel >
        <span value="bird">Bird</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
      </Dropdown>

Dropdown with Default Select Value

    <Dropdown defaultText="Select One ...">
        <span value="bird">Bird</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with Pre-Set Value

    <Dropdown value="cow">
        <span value="bird">Bird</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdowns with Custom Widths:

    <div>
      <div>
        <Dropdown customLabel="Choose" width="300px">
          <span value="bird">Bird</span>
          <span value="cow" selected>Cow</span>
          <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
          <span value="monkey">Monkey</span>
          <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
        </Dropdown>
      </div>
      <br  />
      <div>
        <Dropdown customLabel="Choose" width="7rem">
          <span value="bird">Bird</span>
          <span value="cow" selected>Cow</span>
          <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
          <span value="monkey">Monkey</span>
          <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
        </Dropdown>
      </div>
    </div>

Disabled Dropdown:

    <Dropdown customLabel="Disabled Dropdown" disabled>
       <span value="bird">Bird</span>
       <span value="cow">Cow</span>
       <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
       <span value="monkey">Monkey</span>
       <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Inline Dropdowns:

    <div>
      <Dropdown customLabel="Dropdown One" name="dropdown1" inline>
         <span value="bird">Bird</span>
         <span value="cow">Cow</span>
         <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
         <span value="monkey">Monkey</span>
         <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
      </Dropdown>
      <Dropdown customLabel="Dropdown Two" name="dropdown2" inline>
         <span value="bird">Bird</span>
         <span value="cow">Cow</span>
         <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
         <span value="monkey">Monkey</span>
         <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
      </Dropdown>
    </div>

Dropdown with Required Property:

    <Dropdown defaultText="Select One ..." customLabel="Required Dropdown" required>
       <span value="bird">Bird</span>
       <span value="cow">Cow</span>
       <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
       <span value="monkey">Monkey</span>
       <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with custom validation and error message:

    _errorCallback = (event, value) => {
    	return {isValid: false , message: "Custom error message!"}
    }
    _handleChange = (value, event) => {
      setState({ value: value });
    };
    <Dropdown
      customLabel="Error Message Dropdown"
      errorCallback={_errorCallback}>
       <span value="bird">Bird</span>
       <span value="cow">Cow</span>
       <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
       <span value="monkey">Monkey</span>
       <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with Custom Function Before Change:

    <Dropdown customLabel="onBeforeChange Dropdown" onBeforeChange={ function(value){ let accept = confirm("Do you want to check this?"); return accept; } } >
       <span value="bird">Bird</span>
       <span value="cow">Cow</span>
       <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
       <span value="monkey">Monkey</span>
       <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with onClick function:

    <Dropdown onClick={ function(value, event){ console.log("Click: ", event, value) }}>
       <span value="bird">Bird</span>
       <span value="cow">Cow</span>
       <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
       <span value="monkey">Monkey</span>
       <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Dropdown with onChange function:

    _handleChange = (value, event) => {
      console.log("onChange: ", value, event)
    };
    <Dropdown customLabel="onChange Dropdown"
      onChange={_handleChange}>
        <span value="bird">Bird</span>
        <span value="cow">Cow</span>
        <span value="cbpcdmh">Cow Bird Pick Cat Dog Monkey Human</span>
        <span value="monkey">Monkey</span>
        <span value="eee">eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee</span>
    </Dropdown>

Update Dropdown options via state after initial render:

    initialState = {value: '', items: ["Dog", "Cat"]}
    rows = state.items.map((item, index) => (
      <li value={item} key={index}>
        {item}
      </li>
    ));
    handleToggle = () => {
        setState({  value: '', items: ["Cow", "Horse", "Pig"] })
    }; 
    _handleChange = (value, event) => {
      console.log("onChange: ", value, event)
      setState({ value: value });
    };
    
    <div> 
      <Dropdown defaultText="Select One ..." customLabel="Update State Dropdown" onChange={_handleChange} value={state.value}>
        {rows}
        </Dropdown><br />
        <Button primary onClick={handleToggle}>Update State</Button>
        <br /><br />
        List: {state.items.toString()}
    </div>
