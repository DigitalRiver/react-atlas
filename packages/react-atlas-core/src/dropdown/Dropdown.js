/*
Dropdown inspired and mostly taken from:
react-simple-dropdown
https://github.com/Fauntleroy/react-simple-dropdown
Copyright (c) 2015, Timothy Kempf <tim@kemp59f.info>
*/

import React, { cloneElement, Component, PropTypes } from 'react';
import { classNames } from '../utils';
import DropdownTrigger from './DropdownTrigger';
import DropdownContent from './DropdownContent';
import { findDOMNode } from 'react-dom';
import style from './dropdown.css';

/**
 * Simple Composable Dropdown Component that wraps DropdownTrigger and DropdownContent components. Primarily useful for Navigational dropdowns, not form select dropdowns.
 */
class Dropdown extends Component {
  constructor (props) {
      super(props);
      this.state = {
          active: false
      };
      this.onWindowClick = this.onWindowClick.bind(this);
  }

  onWindowClick(event) {
    // const dropdown_element = findDOMNode( this );
    // if( event.target !== dropdown_element && !dropdown_element.contains( event.target ) && this._isActive() ){
    //   this._hide();
    // }

    if(this._isActive()) {
      this._hide();
    }
  };

  _onToggleClick( event ) {
    event.preventDefault();
    if( this._isActive() ){
      this._hide();
    } else {
      this._show();
    }
  };

  _isActive = () => {
    return ( typeof this.props.active === 'boolean' ) ?
      this.props.active :
      this.state.active;
  };

  _hide() {
    this.setState({
      active: false
    });
    if( this.props.onHide ){
      this.props.onHide();
    }
  }

  _show() {
    this.setState({
      active: true
    });
    if( this.props.onShow ){
      this.props.onShow();
    }
  };

  render () {
    const { children, className, ...props } = this.props;
    // create component classes
    const active = this._isActive();
    const cx = classNames.bind(style);
    var classes = cx(className, {
      container: true
    });

    // stick callback on trigger element
    const bound_children = React.Children.map( children, child => {
      if( child.type === DropdownTrigger ){
        child = cloneElement( child, {
          onClick: this._onToggleClick
        });
      } else if( child.type === DropdownContent ){
        child = cloneElement( child, {
          active
        });
      }
      return child;
    });
    return (
      <div
        {...props} onClick={this.onWindowClick}
        className={classes}
      >
        {bound_children}
      </div>
    );
  };

}

Dropdown.defaultProps = {
  className: ''
};


Dropdown.styleguide = {
  category: 'Navigation',
  index: '5.2',
  wrappedExample: true,
  example: `
// Dropdown Dummy Data {
var countries = [
  { value: 'EN-gb', label: 'England'},
  { value: 'ES-es', label: 'Spain'},
  { value: 'TH-th', label: 'Thailand'},
  { value: 'EN-en', label: 'USA'},
  { value: 'FR-fr', label: 'France'}
];
// }
// Internal Methods {
class DropdownExample extends React.Component {

  handleChange = (dropdown, value) => {
    const newState = {};
    newState[dropdown] = value;
    this.setState(newState);
  };

  render () {
// }
    return (
      <section>
        <h5>Dropdown</h5>
        <p>lorem ipsum...</p>

        <Dropdown>
          <DropdownTrigger>
            <Button>Dropdown Button</Button>
          </DropdownTrigger>
          <DropdownContent>
            <DropdownList>
              {countries.map((country, idx) => (
                <DropdownListItem key={idx}>{country.label}</DropdownListItem>
              ))}
            </DropdownList>
          </DropdownContent>
        </Dropdown>
      </section>
    );
// Mount Component {
  }
}
ReactDOM.render(<DropdownExample/>, mountNode);
// }
`
};

export default Dropdown;
