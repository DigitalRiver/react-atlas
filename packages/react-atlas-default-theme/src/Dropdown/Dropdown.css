.container {
  display: block;
  position: relative;
  z-index: 0;
  margin: 0 .25rem calc(.25rem + 8px) 0;
  top: 8px;
}

.inline {
  display: inline-block;
}

.content {
  composes: clearfix from '../styles.css';
  composes: bg-white from '../styles.css';
  composes: z4 from '../styles.css';
  composes: cursor-pointer from '../styles.css';
  composes: default-font from '../styles.css';
  font-family: Arial, Helvetica, sans-serif;
  font-size: 12px;
  text-align: left;
  left: 0;
  position: relative;
}

.content:hover {
  text-decoration: none;
  border: none;
}

.labelSpacing {
  margin-bottom: 0.35rem;
  padding-right: 5px;
  composes: header-font from '../styles.css';
}

.buttonClass {
  composes: cursor-pointer from '../styles.css';
}

.focus {
  composes: text-input-active from '../styles.css';
}

.list {
  composes: pad-0 from '../styles.css';
  composes: marg-0 from '../styles.css';
  composes: bg-white from '../styles.css';
  composes: default-border-left from '../styles.css';
  composes: default-border-right from '../styles.css';
  composes: default-border-bottom from '../styles.css';
  composes: primary-border-top from '../styles.css';
  composes: rounded-bottom from '../styles.css';
  left: 0;
  top: 1.8rem;
  position: absolute;
  list-style: none;
  transition: height 1s;
  max-height: 250px;
  overflow: auto;
}
.fullWidth{
  min-width: 100px;
  width:100%;
}
.convertedWidth {
  width:calc(100% - 2px);
  min-width:98px;
}
.item {
  padding: 0.5rem;
  -o-text-overflow: ellipsis;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  composes: default-font from '../styles.css';
}
.item:hover {
  text-decoration: none;
  background-color: #e6e6e6;
}
.firstChild {
  margin-top: 5px;
}
.lastChild {
  margin-bottom: 5px;
}
.arrow {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 6px 4px 0 4px;
  border-color: #000000 transparent transparent transparent;
  float: right;
  margin-top: 0.25rem;
  margin-left: 0.25rem;
  position: absolute;
  top: 0.45rem;
  left: calc(100% - 1.1rem);
}

.dropdown-button {
  color: #666;
  border: 1px solid #bebebe;
  border-radius: 2px;
  background-color: white !important;
  text-transform: none !important;
  width: 100%;
  padding-left: 0.5rem;
  text-align: left;
  white-space: nowrap;
  height: 1.77rem;
  line-height: normal;
  composes: default-font from '../styles.css';
}
.dropdown-button:hover {
  border-color: #bebebe;
  background-color: #e6e6e6;
  color: #666;
}
.dropdown-button span {
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 80%;
  float: left;
}

.selected {
  padding: 0.5rem;
  background-color: #ccc;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
.selected:hover {
  background-color: #e6e6e6;
}
.emptyChild {
  height: 0.65rem;
}

.error {
  box-shadow: 0 0 4px #bd0a33;
  border-color: rgba(189, 10, 51, 0.6);
}
.error:hover {
  box-shadow: 0 0 4px #bd0a33;
  border-color: rgba(189, 10, 51, 0.6);
}
.error_message {
  color: #bd0a33;
  margin-bottom: 8px;
  display: inline-block;
  composes: default-font from '../styles.css';
}

.disabledClass {
  opacity: 0.5 !important;
  cursor: default;
}
.disabledClass:hover {
  border: 1px solid #999 !important;
  background-color: white !important;
  outline: none;
}

.requiredIndicator {
  composes: red from '../styles.css';
  margin-left: 5px;
}

.zIndex {
  z-index: 9998;
}

.leftLabel {
  display: table-cell;
  white-space: nowrap;
  vertical-align: middle;
}

.leftLabelContent {
  display: table-cell;
}
