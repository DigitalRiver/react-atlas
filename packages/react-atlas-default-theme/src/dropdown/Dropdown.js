import { Dropdown } from "react-atlas-core";
import style from "./dropdown.css";
import createComponent from "../createComponent";

export default createComponent(Dropdown, style);
