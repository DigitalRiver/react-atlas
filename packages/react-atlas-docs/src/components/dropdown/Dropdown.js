import React, { PropTypes } from "react";
import * as atlas from 'react-atlas';

export default class Dropdown extends React.Component {
    constructor(props){
        super(props)
    }
    render() {
        return (
            <atlas.Dropdown {...this.props}></atlas.Dropdown>
        )
    }
}