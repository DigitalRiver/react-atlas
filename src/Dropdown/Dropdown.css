@import '../styles/variables.css';

.dropdown {
  width: 100%;
  position: relative;
}

.hidden {
  display:none;
}

.options {
  box-sizing: border-box;
  width: 100%;
  border-radius: 0 0 2px 2px;
  border: solid var(--gray);
  border-width: 0 1px 1px 1px;
  margin: -.25rem .25rem .25rem 0;
  position: absolute;
  top: calc(1.96rem);
  z-index: 9998;
  max-height: 250px;
  overflow-y: scroll;
  composes: bg-white from '../styles.css';
}

.leftLabel {
  white-space: nowrap;
}

.fillInput {
  flex: 1;
}

.pointer {
  cursor: pointer;
}

.inline {
  display: inline-block;
  margin-right: 1rem;
}

.arrow {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 6px 4px 0 4px;
  border-color: var(--black) transparent transparent transparent;
  float: right;
  margin-top: 0.25rem;
  margin-left: 0.25rem;
  position: absolute;
  right: 0.5rem;
  top: 0.5rem;
}

.setWidth {
  width: inherit;
  display: inline-block;
}

.arrowUp {
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -o-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}