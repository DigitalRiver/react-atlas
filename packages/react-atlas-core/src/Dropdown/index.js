import { default as DropdownCore } from "./Dropdown";
export { DropdownCore };
