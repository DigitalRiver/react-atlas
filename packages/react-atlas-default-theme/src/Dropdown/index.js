import * as DropdownStyle from "./dropdown.css";
export { DropdownStyle };


