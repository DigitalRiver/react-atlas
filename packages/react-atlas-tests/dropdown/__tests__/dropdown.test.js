import React from "react";
import { mount, shallow, render } from "enzyme";

import renderer from 'react-test-renderer';

import { DropdownCore } from "../../../react-atlas-core/src/Dropdown/index";

function _findItem(n, text) {
  if (n.props().children) {
    if (n.props().children.props) {
      let str = n.props().children.props.value;
      return str == text;
    }
  }
  return false;
}

describe("Test dropdown component", () => {
  it('Test render correctly', () => {
    const comp = <DropdownCore onChange={function() {}}>
					<span>May</span>
					<span value="the">the</span>
					<span value="force">force</span>
					<span value="be">be</span>
					<span value="with">with</span>
					<span value="you">you</span>
				 </DropdownCore>
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test Dropdown component - Mouse tests", () => {
  it("Test Dropdown component - Basic test", function() {
    const component = mount(
      <DropdownCore onChange={function() {}}>
        <span>May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );

    expect(component.state().value).toEqual(null);
    expect(component.state().output).toEqual("May");
  });

  it("Test Dropdown component - Basic test (disabled)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} disabled={true}>
        <span>May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );

    expect(component.state().value).toEqual(null);
    expect(component.state().output).toEqual("May");

    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().active).toEqual(false);

    component.find("Dropdown").simulate("click");
    expect(component.state().output).toEqual("May");
    expect(component.state().active).toEqual(false);
  });

  //	it("Test Dropdown component - Basic test with errorCallback", function(){
  //		const component =  mount(<DropdownCore onChange={ function(){} } errorCallback={ function(ev, val){return {valid: true, message: "Simple text"} } }>
  //									<span value="May">May</span>
  //									<span value="the">the</span>
  //									<span value="force">force</span>
  //									<span value="be">be</span>
  //									<span value="with">with</span>
  //									<span value="you">you</span>
  //								</DropdownCore>);
  //		expect(component.state().value).toEqual('May');
  //		expect(component.state().output).toEqual('May');
  //
  //		component.find('Dropdown').simulate('focus');
  //		expect(component.state().output).toEqual('May');
  //		expect(component.state().focus).toEqual(true);
  //
  //		component.findWhere(n => _findItem(n, 'you')).simulate('mouseDown');
  //	});

  it("Test Dropdown component - Select one item", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);
    component.find("Button").simulate("click");
    expect(component.state().active).toEqual(true);

    component.findWhere(n => _findItem(n, "you")).simulate("mouseDown");

    expect(component.state().value).toEqual("you");
    expect(component.state().index).toEqual(5);

    component.find("Dropdown").simulate("focus");
  });

  it("Test Dropdown component - Select one item - no onChange", function() {
    const component = mount(
      <DropdownCore onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);
    component.find("Button").simulate("click");
    expect(component.state().active).toEqual(true);

    component.findWhere(n => _findItem(n, "you")).simulate("mouseDown");

    expect(component.state().value).toEqual("you");
    expect(component.state().index).toEqual(5);

    component.find("Dropdown").simulate("focus");
  });

  it("Test Dropdown component - Select one item - no onClick", function() {
    const component = mount(
      <DropdownCore onChange={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);
    component.find("Button").simulate("click");
    expect(component.state().active).toEqual(true);

    component.findWhere(n => _findItem(n, "you")).simulate("mouseDown");

    expect(component.state().value).toEqual("you");
    expect(component.state().index).toEqual(5);

    component.find("Dropdown").simulate("focus");
  });

  it("Test Dropdown component - Select one item (twice)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );

    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);
    component.find("Button").simulate("click");
    expect(component.state().active).toEqual(true);

    let item = component.findWhere(n => _findItem(n, "you"));
    item.simulate("mouseDown");
    item.simulate("click");
    expect(component.state().value).toEqual("you");
    expect(component.state().index).toEqual(5);

    component.find("Dropdown").simulate("focus");
    component.find("Button").simulate("click");
    component.findWhere(n => _findItem(n, "May")).simulate("mouseDown");
    expect(component.state().value).toEqual("May");
    expect(component.state().index).toEqual(0);
  });

  it("Test Dropdown component - Click with custon onClick Dropdown", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    component.find("Button").simulate("click");
    component.findWhere(n => _findItem(n, "you")).simulate("mouseDown");
  });

  it("Test Dropdown component - Simple Click on Dropdown with onBeforeChange(false)", function() {
    const component = mount(
      <DropdownCore
        onChange={function() {}}
        onBeforeChange={function() {
          return false;
        }}
      >
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    component.find("Button").simulate("click");
    let item = component.findWhere(n => _findItem(n, "you"));
    item.simulate("mouseDown");
    item.simulate("click");
  });

  it("Test Dropdown component - Simple Click on Dropdown with onBeforeChange(true)", function() {
    const component = mount(
      <DropdownCore
        onChange={function() {}}
        onBeforeChange={function() {
          return true;
        }}
      >
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    component.find("Button").simulate("click");
    let item = component.findWhere(n => _findItem(n, "you"));
    item.simulate("mouseDown");
    item.simulate("click");
  });
});

describe("Test Dropdown component - Keyboard tests", () => {
  it("Test Dropdown component - Select one item (only ArrowDown used)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);

    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "Enter" });

    expect(component.state().value).toEqual("be");
    expect(component.state().output).toEqual("be");
    expect(component.state().index).toEqual(3);
  });

  it("Test Dropdown component - Select one item (ArrowDown & ArrowUp used)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);

    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    component.find("Dropdown").simulate("keyDown", { key: "ArrowUp" });
    component.find("Dropdown").simulate("keyDown", { key: "Enter" });

    expect(component.state().value).toEqual("force");
    expect(component.state().output).toEqual("force");
    expect(component.state().index).toEqual(2);
  });

  it("Test Dropdown component - Select last item (only ArrowDown used)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);

    for (let i = 0; i < 9; i++) {
      component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    }
    component.find("Dropdown").simulate("keyDown", { key: "Enter" });

    expect(component.state().value).toEqual("you");
    expect(component.state().output).toEqual("you");
    expect(component.state().index).toEqual(5);
  });

  it("Test Dropdown component - Select first item (going down & up with ArrowDown & Arrow up)", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);

    for (let i = 0; i < 9; i++) {
      component.find("Dropdown").simulate("keyDown", { key: "ArrowDown" });
    }
    component.find("Dropdown").simulate("keyDown", { key: "Enter" });
    expect(component.state().value).toEqual("you");
    expect(component.state().output).toEqual("you");
    expect(component.state().index).toEqual(5);

    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("you");
    expect(component.state().active).toEqual(true);

    for (let i = 0; i < 9; i++) {
      component.find("Dropdown").simulate("keyDown", { key: "ArrowUp" });
    }
    component.find("Dropdown").simulate("keyDown", { key: "Enter" });
    expect(component.state().value).toEqual("May");
    expect(component.state().output).toEqual("May");
    expect(component.state().index).toEqual(0);
  });

  it("Test Dropdown component - Press unhandled key", function() {
    const component = mount(
      <DropdownCore onChange={function() {}} onClick={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    component.find("Dropdown").simulate("focus");
    expect(component.state().output).toEqual("May");
    expect(component.state().focus).toEqual(true);

    component.find("Dropdown").simulate("keyDown", { key: "Ctrl" });
  });
});

describe("Test Dropdown component - Window blur tests", () => {
  it("Test Dropdown component - onBlur event test", function() {
    const component = mount(
      <DropdownCore onChange={function() {}}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );

    expect(component.state().value).toEqual("May");
    expect(component.state().output).toEqual("May");
    component.find("Dropdown").simulate("focus");
    component.find("Dropdown").simulate("blur");
  });
});

describe("Test Dropdown component - Dropdown Regression tests", () => {
  it("Regression test for bug #249", function() {
    expect(function() {
      const component = mount(<DropdownCore />);
    }).toThrow(
      new Error("You must pass at least one child component to Dropdown")
    );
    // expect.assertions(0);
  });
});

describe("Test Dropdown component - Dropdown Regression tests", () => {
  it("Regression test for bug #405", function() {
    const component = mount(
      <DropdownCore isValid={false}>
        <span value="May">May</span>
        <span value="the">the</span>
        <span value="force">force</span>
        <span value="be">be</span>
        <span value="with">with</span>
        <span value="you">you</span>
      </DropdownCore>
    );
    expect(component.state().isValid).toEqual(false);
  });
});
