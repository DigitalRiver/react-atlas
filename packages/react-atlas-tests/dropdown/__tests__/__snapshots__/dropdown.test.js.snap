// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test dropdown component - Basic Test render correctly 1`] = `
<div
  className=""
  name={undefined}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  style={undefined}
  styleName="container"
>
  <div
    style={
      Object {
        "minWidth": "100px",
        "width": "100%",
      }
    }
    styleName="content"
  >
    <div
      styleName="fullWidth"
    >
      <button
        className=""
        href={undefined}
        onClick={[Function]}
        style={undefined}
        styleName="button default_btn"
        type="button"
      >
        <span>
          May
        </span>
        <i
          styleName="arrow"
        />
      </button>
    </div>
    <input
      type="hidden"
      value="may"
    />
  </div>
</div>
`;
